pkg/backup/item_backupper.go

```diff
@@ -253,11 +253,34 @@ func (ib *defaultItemBackupper) backupItem(logger logrus.FieldLogger, obj runtim
 		return false, kubeerrs.NewAggregate(backupErrs)
 	}
 
+	// group version of this object
+	// Used on filepath to backup up all groups and versions
+	version := resourceVersion(obj)
+
+	// Getting the preferred group version of this resource
+	gvr_preferred, _, err := ib.discoveryHelper.ResourceFor(groupResource.WithVersion(""))
```

Comment:
To clarify my understanding of this call - since we're invoking `ResourceFor` without any version, we'll get back the preferred GVK, and thus we can use the version on that GVK as preferred.

If we invoked `ResourceFor` with a specific version, we'd get the GVK for that specific version, assuming it existed.

And the reason `ResourceFor` is used here is because the helper for `ServerPreferredResources` gets us a list of preferred versions for _all_ API groups, not just the one we're processing at the moment.

Is that accurate?

Reply:
Yes, your understanding is correct.